contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```
@@ -606,7 +621,12 @@ resolve_numeric_typmod_from_exp(Plan *plan, Node *expr)
 						tle = get_tle_by_resno(subplan->targetlist, var->varattno);
 						if (!tle)
 							elog(ERROR, "bogus varattno for SubqueryScan's subplan: %d", var->varattno);
-						return resolve_numeric_typmod_from_exp(subplan, (Node *)tle->expr);
+						rettypmod = resolve_numeric_typmod_from_exp(subplan, (Node *)tle->expr, &found_typmod);
+						if (!found_typmod)
+						{
+							if (found != NULL) *found = false;
+						}
+						return rettypmod;
 					}
 					else
```

Do we have such errors in the engine as well? can be an assert right? How is subplan null in SubqueryScan

---

Same for MergeAppend

---

Existing code. Original Author @tanyagupta17 can help answer this. PR: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3457 which added this code.

---

Please refer to this thread https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3457#discussion_r1973714236
This was a safety check, as assertion does not work in production, it was recommended by @tanscorpio7 to add these sanity checks.

---

Should we really throw an error here? , Not sure if this helps User in any way?